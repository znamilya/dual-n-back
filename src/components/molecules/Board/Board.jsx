import React, { PropTypes } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import bemCl from 'bem-cl';

import './Board.styl';


const b = bemCl('board');

class Board extends React.PureComponent {

    static propsTypes = {
        position: PropTypes.number,
        step: PropTypes.number,
    };

    constructor(props) {
        super(props);

        this.cellsStub = this.createCells();
    }


    /* ------------------------------------------------------------------------------------------ */
    /* METHODS                                                                                    */
    /* ------------------------------------------------------------------------------------------ */
    createCells() {
        return (new Array(9))
            .fill(1)
            .map((cell, index) => {
                return {
                    id: `cell-${index}`,
                };
            });
    }


    /* ------------------------------------------------------------------------------------------ */
    /* RENDER                                                                                     */
    /* ------------------------------------------------------------------------------------------ */
    renderCells() {
        return this.cellsStub.map((cell, index) => (
            <li className={b('cell', {
                active: index === this.props.position
            })}
                key={cell.id + this.props.step}
            />
        ));
    }

    render() {
        return (
            <div className={b()}>
                <ul className={b('cells')}>
                    {this.renderCells()}
                </ul>
            </div>
        );
    }
}


export default Board;
