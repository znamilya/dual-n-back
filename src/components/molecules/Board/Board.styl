.board
    position: relative

    width: 100%
    padding: 100% 0 0

    background: tomato


    &__cells
        position: absolute
        top: 0
        right: 0
        bottom: 0
        left: 0

        margin: 0
        padding: 0

        list-style: none


    &__cell
        box-sizing: border-box
        float: left

        width: calc(100% / 3)
        padding: calc(100% / 3) 0 0

        box-shadow: inset 0 0 0 1px #000


        &_active
            background: yellowgreen


.cell-appear
    opacity: 0

.cell-appear.cell-appear-active
    opacity: 1
    transition: opacity .5s ease-in
