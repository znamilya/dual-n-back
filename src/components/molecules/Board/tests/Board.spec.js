import React from 'react';

import Board from '../Board';


describe('<Board />', () => {
    it('should render itself', () => {
        const wrapper = shallow(
            <Board
                step={1}
                position={1}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should add active class to active cell', () => {
        const position = 1;
        const wrapper = shallow(
            <Board
                step={1}
                position={position}
            />
        );

        expect(wrapper.find('.board__cell').at(position).hasClass('board__cell_active')).toBe(true);
    });
});
