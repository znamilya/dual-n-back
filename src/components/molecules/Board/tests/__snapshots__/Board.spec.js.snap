exports[`<Board /> should render itself 1`] = `
<div
  className="board">
  <ul
    className="board__cells">
    <li
      className="board__cell" />
    <li
      className="board__cell board__cell_active" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
    <li
      className="board__cell" />
  </ul>
</div>
`;
